### PR TITLE
Add accessbility image / video text

### LINKED ISSUE #441: Video - Options
## Feature Description

As an editor utilizing videos, I want to see what options are available for my videos

**Waiting for Feature Brief**

**https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=2671%3A55920**

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

1. The editor is presented with a looping option
2. The editor is able to mute a video (the video will have no sound when played) (Note: Videos are muted in AMP)

### LINKED ISSUE #486: Assistive Text Options for Media
## Feature Description

This was pulled out of #441 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
1. The editor is presented with assisted text options.

### PR DESCRIPTION
**Pull request by spacedmonkey**
Add alt text to video and image elements. 

Fixes #486.

### CODE REVIEW

File: assets/src/edit-story/components/panels/videoAccessibility.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        <ExpandedTextInput
+          placeholder={__('Assistive text', 'web-stories')}
+          value={state.alt || ''}
+          onChange={handleChange('alt')}
+          clear
+        />
+      </Row>
+      <Row>
+        <Note>
+          {__('Enter an address to apply a 1 or 2 tap link', 'web-stories')}
```
Comment (swissspidy) [inline assets/src/edit-story/components/panels/videoAccessibility.js]: Copy/paste issue?

Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/videoAccessibility.js]: Fixed in https://github.com/google/web-stories-wp/pull/652/commits/0f570ebc94f3f3b1b1267a4bd076387590611813

File: assets/src/edit-story/components/panels/index.js [hunk 2]
```diff
@@ -84,7 +87,10 @@ export function getPanels(elements) {
     ];
     // If the selected element's type is video, display poster panel, too.
```
Comment (swissspidy) [inline assets/src/edit-story/components/panels/index.js]:
```suggestion
    // If the selected element's type is video, display accessibility panel, too.
```

File: assets/src/edit-story/elements/image/index.js [hunk 3]
```diff
@@ -42,4 +42,8 @@ export const defaultAttributes = {
   ...MEDIA_DEFAULT_ATTRIBUTES,
 };
 
-export const panels = [...MEDIA_PANELS, PanelTypes.LINK];
+export const panels = [
+  ...MEDIA_PANELS,
+  PanelTypes.IMAGE_ACCESSIBILITY,
+  PanelTypes.LINK,
```
Comment (swissspidy) [inline assets/src/edit-story/elements/image/index.js:47]: According to Figma, the accessibility panel should come after the link panel.

Right now, the accessbility panel seems to be the very first one.

File: assets/src/edit-story/elements/video/index.js [hunk 4]
```diff
@@ -46,11 +46,12 @@ export const defaultAttributes = {
...
     poster: null,
     videoId: 0,
+    title: '',
   },
 };
 
 export const panels = [
   ...MEDIA_PANELS,
-  PanelTypes.VIDEO_POSTER,
+  PanelTypes.VIDEO_ACCESSIBILITY,
```
Comment (swissspidy) [inline assets/src/edit-story/elements/video/index.js]: According to Figma, the accessibility panel should come after the link panel.

File: assets/src/edit-story/components/panels/index.js [hunk 5]
```diff
@@ -34,27 +35,29 @@ export { default as ColorPresetPanel } from './colorPreset';
...
-const VIDEO_POSTER = 'videoPoster';
+const VIDEO_ACCESSIBILITY = 'videoAccessibility';
 const PAGE = 'page';
 const NO_SELECTION = 'noselection';
 
 export const PanelTypes = {
   BACKGROUND_SIZE_POSITION,
   BACKGROUND_DISPLAY,
   BACKGROUND_OVERLAY,
+  IMAGE_ACCESSIBILITY,
```
Comment (swissspidy) [inline assets/src/edit-story/components/panels/index.js]: This should probably move to the bottom, as per the order in Figma

File: assets/src/edit-story/components/panels/index.js [hunk 6]
```diff
@@ -34,27 +35,29 @@ export { default as ColorPresetPanel } from './colorPreset';
...
   SIZE_POSITION,
   STYLE,
   LAYER_STYLE,
   TEXT,
   TEXT_STYLE,
   LINK,
-  VIDEO_POSTER,
+  VIDEO_ACCESSIBILITY,
```
Comment (swissspidy) [inline assets/src/edit-story/components/panels/index.js:60]: There's some shared code between `VideoAccessibilityPanel` and `ImageAccessibilityPanel`. Wouldn't it make sense to have 1 accessibility panel for both?

File: assets/src/edit-story/components/panels/index.js (hunk 5 above)
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/index.js]: Fixed in https://github.com/google/web-stories-wp/pull/652/commits/eef36fbe3357f8343b08bf9f4daa513550c45378

File: assets/src/edit-story/elements/video/index.js (hunk 4 above)
Comment (spacedmonkey) [reply assets/src/edit-story/elements/video/index.js]: Fixed in https://github.com/google/web-stories-wp/pull/652/commits/eef36fbe3357f8343b08bf9f4daa513550c45378

File: assets/src/edit-story/elements/image/index.js (hunk 3 above)
Comment (spacedmonkey) [reply assets/src/edit-story/elements/image/index.js:47]: Fixed in https://github.com/google/web-stories-wp/pull/652/commits/eef36fbe3357f8343b08bf9f4daa513550c45378

File: assets/src/edit-story/components/panels/index.js (hunk 6 above)
Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/index.js:60]: There is only one shared field between the video and image. So for that reason, I didnt see the point, as the complexity added to make it shared outweighs the benefit.

File: assets/src/edit-story/components/panels/index.js [hunk 7]
```diff
@@ -82,9 +85,12 @@ export function getPanels(elements) {
       { type: BACKGROUND_DISPLAY, Panel: BackgroundDisplayPanel },
       { type: BACKGROUND_OVERLAY, Panel: BackgroundOverlayPanel },
     ];
-    // If the selected element's type is video, display poster panel, too.
+    // If the selected element's type is video, display accessibility panel, too.
```
Comment (swissspidy) [inline assets/src/edit-story/components/panels/index.js]: Shouldn't the accessibility panel also be shown for background images?

Comment (spacedmonkey) [reply assets/src/edit-story/components/panels/index.js]: Fixed in https://github.com/google/web-stories-wp/pull/652/commits/fb9393afca669dbd67eef52a3ca727c8d9a54de0

**Review by swissspidy — APPROVED** [review]

**Pull request merged**